src/conditional.jl

```diff
@@ -195,11 +195,11 @@ function grad_apply_sparsity!(rbm::ConditionalRBM, X::Mat,
                   throw(ArgumentError("If :sparsity_cost is used, :sparsity_target should also be defined")))
     curr_sparsity = mean(hid_means(rbm, vis))
     penalty = cost * (curr_sparsity - target)
-    axpy!(-penalty, dW, dW)
-    axpy!(-penalty, dA, dA)
-    axpy!(-penalty, dB, dB)
-    axpy!(-penalty, db, db)
-    axpy!(-penalty, dc, dc)
+    axpy!(-penalty, ones(size(dW)), dW)
```

Comment:
Oh, yeah, good point. The problem with `dW -= penalty` is that it will change the variable binding for `dW` rather than changing the initial contents of `dW`, however, it might make sense to just have an `add!` function which changes the contents.